optimade/adapters/structures/aiida.py

```diff
@@ -45,13 +44,13 @@ def get_aiida_structure_data(optimade_structure: OptimadeStructure) -> Structure
     attributes = optimade_structure.attributes
 
     # Convert null/None values to float("nan")
-    lattice_vectors, adjust_cell = pad_cell(attributes.lattice_vectors)
+    lattice_vectors, adjust_cell = pad_cell(attributes.lattice_vectors)  # type: ignore[arg-type]
     structure = StructureData(cell=lattice_vectors)
 
     # If species not provided, infer data from species_at_sites
     species: Optional[List[OptimadeStructureSpecies]] = attributes.species
     if not species:
-        species = species_from_species_at_sites(attributes.species_at_sites)
+        species = species_from_species_at_sites(attributes.species_at_sites)  # type: ignore[arg-type]
```

Comment:
The type of the argument in species_from_species_at_sites is List[str] and the type of species_at_sites is List[str] so the `#type: ignore` should not be necessary here. (I do not have Aiida so I could not test this.)

Reply:
It's `Optional[List[str]]` which is why mypy flags it up:

```
optimade/adapters/structures/aiida.py:53: error: Argument 1 to "species_from_species_at_sites" has incompatible type "Optional[List[str]]"; expected "List[str]"  [arg-type]
```

Reply:
Thanks for the clarification. I did not know mypy was that strict.